webclient/services/mailer.py

```diff
@@ -518,25 +521,28 @@ def send_en02(cls, send_to, project_ids):
     def send_en03_en04(cls, project: 'pas.models.SamostatnyNalez', reason):
         IDENT_CELY = 'E-N-03'
         logger_s.debug("services.mailer.send", ident_cely=IDENT_CELY)
+        states = {3: POTVRZENI_SN, 4: ARCHIVACE_SN}
         notification_type = uzivatel.models.UserNotificationType.objects.get(ident_cely=IDENT_CELY)
         subject = notification_type.predmet.format(ident_cely=project.ident_cely)
-        look_for_stav = ARCHIVACE_SN
-        if project.stav == POTVRZENI_SN:
-            look_for_stav = POTVRZENI_SN
-        if project.stav != ARCHIVACE_SN or project.stav != POTVRZENI_SN:
+        look_for_stav = 4
+        if project.stav == 3:
+            look_for_stav = 3
+        if project.stav != 4 and project.stav != 3:
             return
         log = Historie.objects.filter(
-            vazba__sn_historie__ident_cely=project.ident_cely, typ_zmeny=look_for_stav
+            vazba__sn_historie__ident_cely=project.ident_cely, typ_zmeny=states[look_for_stav]
         ).order_by('datum_zmeny').first()
         if not log:
             return
-        user = log.uzivatel.email
+        user = log.uzivatel
         html = render_to_string(notification_type.cesta_sablony, {
             "title": subject,
             "katastr": project.katastr.nazev,
             "reason": reason,
-            "state": project.PAS_STATES[project.stav][1],
-            "server_domain": settings.EMAIL_SERVER_DOMAIN_NAME
+            "state": project.PAS_STATES[project.stav - 2][1],
```

Comment:
Having project.stav - 2 is a little bit obscure. Can't it be replaced by some named constant? It is not obvious why "-2"